src/modules/job-manager/jobtap.c

```diff
@@ -0,0 +1,520 @@
+                                               const char *fmt,
+                                               va_list ap)
+{
+    flux_plugin_arg_t *args = flux_plugin_arg_create ();
+    if (!args)
+        return NULL;
+
+    if (flux_plugin_arg_pack (args,
+                              FLUX_PLUGIN_ARG_IN,
+                              "{s:O s:I s:i s:i s:i s:I s:f}",
+                              "jobspec", job->jobspec_redacted,
+                              "id", job->id,
+                              "userid", job->userid,
+                              "urgency", job->urgency,
+                              "state", job->state,
+                              "priority", job->priority,
+                              "t_submit", job->t_submit) < 0)
+        goto error;
+
+    if (fmt
+        && flux_plugin_arg_vpack (args,
+                                  FLUX_PLUGIN_ARG_IN|FLUX_PLUGIN_ARG_UPDATE,
+                                  fmt, ap) < 0)
+        goto error;
+
+    /*
+     *  Always start with empty OUT args. This allows unpack of OUT
+     *   args to work without error, even if plugin does not set any
+     *   OUT args.
+     */
+    if (flux_plugin_arg_set (args, FLUX_PLUGIN_ARG_OUT, "{}") < 0)
+        goto error;
+
+    return args;
+error:
+    flux_plugin_arg_destroy (args);
+    return NULL;
+}
+
+int jobtap_get_priority (struct jobtap *jobtap,
+                         struct job *job,
+                         int64_t *pprio)
+{
+    int rc = -1;
+    flux_plugin_arg_t *args;
+    va_list ap;
+    int64_t priority = -1;
+
+    if (!jobtap || !job || !pprio) {
+        errno = EINVAL;
+        return -1;
+    }
+    if (!jobtap->plugin) {
+        *pprio = job->urgency;
+        return 0;
+    }
+    /*  N.B.: passing uninitialized ap to args_vcreate, but it should be
+     *   ok, since fmt == NULL, ap will not be used.
+     */
+    if (!(args = jobtap_args_vcreate (jobtap, job, NULL, ap)))
+        return -1;
+    rc = flux_plugin_call (jobtap->plugin, "job.priority.get", args);
+    if (rc == 1) {
+        /*
+         *  A priority.get callback was run. Try to unpack a new priority
+         */
+        if (flux_plugin_arg_unpack (args, FLUX_PLUGIN_ARG_OUT,
+                                    "{s?I}",
+                                    "priority", &priority) < 0) {
+            flux_log (jobtap->ctx->h, LOG_ERR,
+                      "jobtap: %s: job.priority.get: arg_unpack: %s",
+                      jobtap_plugin_name (jobtap->plugin),
+                      flux_plugin_arg_strerror (args));
+            rc = -1;
+        }
+        if (priority == -1) {
+           /*
+            *  A plugin callback was called but didn't provide a
+            *   priority. This could be due to a loaded plugin that is
+            *   not a priority plugin. Therefore take the default action
+            *   and set priority to job->urgency.
+            */
+            priority = job->urgency;
```

Comment:
>  So returning FLUX_JOBTAP_PRIORITY_DEFAULT informs the job-manager to do its default thing for this job.

￼
That is what just not returning a priority in the plugin OUT args currently does. I don't think there needs to be another way to do it.

Reply:
I guess I view "I want the job-manager to do the default" vs "I did not to set a priority" as different.  Like the latter could have been a mistake from the plugin writer?  But overall this isn't a huge deal.

Reply:
At this point, there is no way to tell if the plugin is a priority-managing plugin or not. I'd like to add support for loading multiple plugins in the future, and at that point it may make sense to separate the "priority" plugin from more general plugins, at which time maybe we'll be able to call out an error for priority plugins which do not set a priority. At that time it indeed may make sense for the plugin to return an "unset" priority (meaning take default action) so as to avoid the error. So it is a good idea.